abci/ve/vote_extension.go

```diff
@@ -76,47 +77,51 @@ func (h *VoteExtensionHandler) ExtendVoteHandler() sdk.ExtendVoteHandler {
 	return func(ctx sdk.Context, req *cometabci.RequestExtendVote) (resp *cometabci.ResponseExtendVote, err error) {
 		start := time.Now()
 
-		// measure latencies from invocation to return
+		// measure latencies from invocation to return, catch panics first
 		defer func() {
+			// catch panics if possible
+			if r := recover(); r != nil {
+				h.logger.Error(
+					"recovered from panic in ExtendVoteHandler",
+					"err", r,
+				)
+
+				resp, err = &cometabci.ResponseExtendVote{VoteExtension: []byte{}}, Panic{fmt.Errorf("%v", r)}
+			}
+
+			// measure latency
 			latency := time.Since(start)
 			h.logger.Info(
 				"extend vote handler",
 				"duration (seconds)", latency.Seconds(),
 				"err", err,
 			)
-			h.metrics.ObserveABCIMethodLatency(servicemetrics.ExtendVote, latency)
+			slinkyabci.RecordLatencyAndStatus(h.metrics, latency, err, servicemetrics.ExtendVote)
```

Comment:
Is there a reason you're introducing returned errors in this function? The [SDK](https://github.com/cosmos/cosmos-sdk/blob/927ac38dc708bda4efb66cffa643b09697babc38/baseapp/abci.go#L632) catches them and returns `nil` like we are currently doing.

Seems like this will change the behavior of the oracle from defaulting to an empty vote extension on failure to failing to add a vote extension at all. It looks like it will [get swallowed](https://github.com/cometbft/cometbft/blob/635d0b596c7300c98caa57e7aab26a7a4579ab97/internal/consensus/state.go#L2486) by comet, so maybe it nets out to the same outcome in the end?

Reply:
yes, for each ABCI method, comet's behavior is to eventually panic on errors returned from the app. Hence, the baseapp's implementations in `abci.go` log and ignore errors from our handlers. Hence, I thought it made sense to keep our logic in line w/ our other handlers. For instance, Prepare / ProcessProposal return errors in execution to the base-app despite ultimately being swallowed